### PR TITLE
Feat: Add ledger legacy migration transaction popup

### PR DESCRIPTION
**Pull request by begonaalvarezd**
# Description of change

_**Important: legacy checksum is giving a validation error so its being commented for now for the sake of the build**_

Add legacy migration transaction popup 
Screenshot from 2021-07-14 15-53-40 Screenshot from 2021-07-14 15-54-05

_**Note: this PR depends on #1246. When #1246 gets merged, the base of this PR can be changed to `feat/ledger`**_

## Links to any relevant issues

N/A

## Type of change

- New (a change which implements a new feature)

## How the change has been tested

Ubuntu 18.04. 
Ledger Nano S, firmware 2.0.0

Migration flows tested:
- [x] Smooth migration (happy)
- [x] Fragmented funds 

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: packages/shared/components/popups/LedgerLegacyTransaction.svelte [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        balance: number
+        index: number
+        checksum: string
+    }
+
+    export let locale
+    export let output: Output
+    export let inputs: Input[]
+
+    const formatAmount = (amountRaw: number): string => {
```
Comment (laumair) [inline packages/shared/components/popups/LedgerLegacyTransaction.svelte:21]: I see a similar method here https://github.com/iotaledger/firefly/pull/1246/files#diff-779dc1ec1c8db9e429fe726b464b33b7e906c463e66ee937c786127d40a3f7d8R55-R57. I think it is best to have it merged or maybe the abstraction over the `formatUnitBestMatch` is redundant.

File: packages/shared/components/popups/LedgerLegacyTransaction.svelte [hunk 2]
```diff
@@ -0,0 +1,58 @@
+<script lang="ts">
+    import { Illustration, Text } from 'shared/components'
+    import { formatUnitBestMatch } from 'shared/lib/units'
+
+    type Output = {
+        address: string
+        value: number
+        checksum: string
+    }
+    type Input = {
```
Comment (laumair) [inline packages/shared/components/popups/LedgerLegacyTransaction.svelte:10]: Best to define these types in `shared`. Could possibly be reused [here](https://github.com/iotaledger/firefly/pull/1246/files#diff-502cfa0c2543d63e9efdbfe5989c4b55564c61a34cd305316f94a6216ac5bc0cR146)

File: packages/shared/lib/migration.ts [hunk 3]
```diff
@@ -1330,4 +1332,17 @@ export const initialiseMigrationListeners = () => {
     })
 }
 
+function openLedgerLegacyTransactionPopup(_transfer, _inputs) {
```
Comment (laumair) [inline packages/shared/lib/migration.ts:1335]: Should probably define the types (& return type too) for this function.

File: packages/shared/lib/migration.ts [hunk 4]
```diff
@@ -1330,4 +1332,17 @@ export const initialiseMigrationListeners = () => {
...
+    const inputs = _inputs.map(_input => ({ ..._input, checksum: getChecksum(_input.address) }))
 
+    openPopup({
+        type: 'ledgerLegacyTransaction',
+        hideClose: true,
+        props: {
+            output, inputs
+        }
+    })
+}
```
Comment (laumair) [inline packages/shared/lib/migration.ts:1348]: Missing newline

**Review by laumair — APPROVED** [review]

**Review by laumair — CHANGES_REQUESTED** [review]
Looks good overall.

File: packages/shared/components/popups/LedgerLegacyTransaction.svelte (hunk 2 above)
Comment (begonaalvarezd) [reply packages/shared/components/popups/LedgerLegacyTransaction.svelte:10]: These types are not exactly shareable, checksum for example is not part of the normal input/output. But I'll give it a thought, thanks!

File: packages/shared/components/popups/LedgerLegacyTransaction.svelte (hunk 1 above)
Comment (begonaalvarezd) [reply packages/shared/components/popups/LedgerLegacyTransaction.svelte:21]: Redundant, removing

**Pull request merged**